spark/v3.2/spark/src/main/scala/org/apache/spark/sql/execution/datasources/SparkExpressionConverter.scala

```diff
@@ -30,4 +34,18 @@ object SparkExpressionConverter {
     // But these two conversions already exist and well tested. So, we are going with this approach.
     SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
   }
+
+  @throws[AnalysisException]
+  def collectResolvedSparkExpression(session: SparkSession, tableName: String, where: String): Expression = {
+    var expression:Expression = null
+    // Add a dummy prefix linking to the table to collect the resolved spark expression from analyzed plan.
+    val prefix = String.format("SELECT 42 from %s where ", tableName)
+    val logicalPlan = session.sessionState.sqlParser.parsePlan(prefix + where)
+    val analyzedLogicalPlan = session.sessionState.executePlan(logicalPlan, CommandExecutionMode.ALL).analyzed
+    analyzedLogicalPlan.collectFirst {
+      case filter: Filter =>
+        expression = filter.expressions.head
+    }
+    expression
```

Comment:
One question here, since we are pulling out the relation node here to get these filters .. do we still need
```convertToIcebergExpression```?

Since we basically can just call
```
SparkFilters.convert(analyzedLogicalPlan.collectFirst { case filter: Filter =>  expression = filter.expressions.head }) 
```
?

Also is ok to just use collectFirst? Don't we need all possible Filter expressions? I may have forgotten how this works.

Reply:
a. we can't use SparkFilters.convert, as Filter in plan is `org.apache.spark.sql.catalyst.plans.logical.Filter` and Filter in SparkFilters.convert is `org.apache.spark.sql.sources.Filter`

b. I think we can use collectFirst as the plan is fixed always as query is fixed. It is simple Project + Filter + scan operators in the pan.